File: Source/Affixes/Unpredictable.cs
using static WCSharp.Api.Common;

public class Unpredictable : Affix
{
    private const int AFFIX_ABILITY = Constants.ABILITY_UNPREDICTABLE;
    private static int WANDER_ABILITY = FourCC("Awan");

    public Unpredictable(Wolf unit) : base(unit)
    {
        Name = "Unpredictable";
    }

    public override void Apply()
    {
        Unit.Unit.AddAbility(WANDER_ABILITY); // Wander
        Unit.Unit.AddAbility(AFFIX_ABILITY);
        Unit.OverheadEffectPath = "";
        base.Apply();
    }

    public override void Remove()
    {
        Unit.Unit.RemoveAbility(WANDER_ABILITY); // Wander
        Unit.Unit.RemoveAbility(AFFIX_ABILITY);
        Unit.OverheadEffectPath = Wolf.DefaultOverheadEffect;
        base.Remove();
    }

    public override void Pause(bool pause)
    {

    }

}

File: src/Affixes/Unpredictable.ts
export class Unpredictable extends Affix {
    private AFFIX_ABILITY: number = Constants.ABILITY_UNPREDICTABLE
    private static WANDER_ABILITY: number = FourCC('Awan')

    public constructor(unit: Wolf) {
        super(unit)
        Name = 'Unpredictable'
    }

    public override Apply() {
        UnitAddAbility(this.Unit.Unit, this.WANDER_ABILITY) // Wander
        UnitAddAbility(this.Unit.Unit, this.AFFIX_ABILITY)
        Unit.OVERHEAD_EFFECT_PATH = ''
        super.Apply()
    }

    public override Remove() {
        UnitRemoveAbility(this.Unit.Unit, this.WANDER_ABILITY) // Wander
        UnitRemoveAbility(this.Unit.Unit, this.AFFIX_ABILITY)
        Unit.OVERHEAD_EFFECT_PATH = Wolf.DEFAULT_OVERHEAD_EFFECT
        super.Remove()
    }

    public override Pause(pause: boolean) {}
}
